Foundation/FileHandle.swift

```diff
@@ -47,6 +60,79 @@ open class FileHandle : NSObject, NSSecureCoding {
     }
 
     internal func _readDataOfLength(_ length: Int, untilEOF: Bool, options: NSData.ReadingOptions = []) throws -> NSData.NSDataReadResult {
+#if os(Windows)
+        if _handle == INVALID_HANDLE_VALUE {
+          return NSData.NSDataReadResult(bytes: nil, length: 0, deallocator: nil)
+        }
+        if length == 0 && !untilEOF {
+          // Nothing requested, return empty response
+          return NSData.NSDataReadResult(bytes: nil, length: 0, deallocator: nil)
+        }
+
+        var fiFileInfo: BY_HANDLE_FILE_INFORMATION = BY_HANDLE_FILE_INFORMATION()
+        if GetFileInformationByHandle(_handle, &fiFileInfo) == FALSE {
+          throw NSError(domain: NSPOSIXErrorDomain, code: Int(GetLastError()),
+                        userInfo: nil)
+        }
+
+        if fiFileInfo.dwFileAttributes & DWORD(FILE_ATTRIBUTE_NORMAL) == FILE_ATTRIBUTE_NORMAL {
+          if options.contains(.alwaysMapped) {
+            let hMapping: HANDLE =
+                CreateFileMappingA(_handle, nil, DWORD(PAGE_READONLY), 0, 0, nil)
+            if hMapping == HANDLE(bitPattern: 0) {
+              fatalError("CrateFileMappingA failed")
+            }
+
+            let szFileSize: UInt64 = (UInt64(fiFileInfo.nFileSizeHigh) << 32) | UInt64(fiFileInfo.nFileSizeLow << 0)
+            let szMapSize: UInt64 = Swift.min(UInt64(length), szFileSize)
+            let pData: UnsafeMutableRawPointer =
+                MapViewOfFile(hMapping, DWORD(FILE_MAP_READ), 0, 0, szMapSize)
+
+            return NSData.NSDataReadResult(bytes: pData, length: Int(szMapSize)) { buffer, length in
+              if UnmapViewOfFile(buffer) == FALSE {
+                fatalError("UnmapViewOfFile failed")
+              }
+              if CloseHandle(hMapping) == FALSE {
+                fatalError("CloseHandle failed")
+              }
+            }
+          }
+        }
+
+        let blockSize: Int = 8 * 1024
+        var allocated: Int = blockSize
+        var buffer: UnsafeMutableRawPointer = malloc(allocated)!
+        var total: Int = 0
+
+        defer { free(buffer) }
```

Comment:
The buffer passedd to the `NSData.NSDataReadResult()` call at the bottom isnt copied so this buffer shouldnt be freed at function exit.